Kernel/Arch/x86/CPU.h

```diff
@@ -1003,6 +1005,21 @@ class Processor {
     static Vector<FlatPtr> capture_stack_trace(Thread& thread, size_t max_frames = 0);
 
     String platform_string() const;
+
+    ALWAYS_INLINE void enter_profiler()
+    {
+        ++m_nested_profiler_calls;
```

Comment:
Can we please make this a static function and then doing read/writes using the fs register with e.g. `read_fs_u32`/`write_fs_u32` (I have a u8 variant in https://github.com/SerenityOS/serenity/pull/5184/commits/e5bed4e54153ca7c95eab94cef1ca38e228059de as well). This would close a window for calling `Processor::current()` and calling this function where a thread could have migrated to another CPU. I converted a lot of the Processor functions to do this in #5184 because it created a lot of weird problems. I'm not sure if you can do an increment using the fs register, though, if you can't do it atomically then you'll have to use a `InterruptDisable` and call Processor::current() once interrupts are disabled.

Reply:
I've moved the `m_nested_profiler_calls` field to the `Thread` class because even if we're preempted and resume on another processor this attribute really belongs to the call stack of the current thread. I've also removed the `InterruptDisabler`. What's your take on this idea? :)